ui/raidboss/data/06-ew/ultimate/dragonsongs_reprise_ultimate.ts

```diff
@@ -818,14 +827,306 @@ const triggerSet: TriggerSet<Data> = {
+              return output.elusiveTower3!();
+          }
+
+          // Call Tower 2 Soak (based on previous position)
+          if (num === 1) {
+            if (data.diveFromGracePreviousPosition[data.me] === 'west')
+              return output.spineshatterTower2!();
+            if (data.diveFromGracePreviousPosition[data.me] === 'east')
+              return output.elusiveTower2!();
+          }
+        }
+        // Third Dive, on num3s
+        if (data.diveFromGraceTowerCounter === 3) {
+          // Num3s need to move out of tower 3
+          if (num === 3)
+            return output.move!();
+        }
+      },
+      run: (data) => {
+        // Empty the collector
+        for (const player in data.diveFromGracePositions)
+          delete data.diveFromGracePositions[player];
+      },
+      outputStrings: {
+        move: Outputs.moveAway,
+        stackNorth: {
+          en: 'Stack North',
+        },
+        spineshatterTower2: {
+          en: 'Backright Tower',
+        },
+        spineshatterTower3: {
+          en: 'Stack => Right Tower',
+        },
+        elusiveTower2: {
+          en: 'Backleft Tower',
+        },
+        elusiveTower3: {
+          en: 'Stack => Left Tower',
+        },
+      },
+    },
+    {
+      id: 'DSR Dive From Grace Tower 1',
+      // Triggered on first instance of Eye of the Tyrant (6714)
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '6714', source: 'Nidhogg', capture: false }),
+      // Ignore targetIsYou() incase player misses stack
+      condition: (data) => data.eyeOfTheTyrantCounter === undefined,
+      suppressSeconds: 1,
+      infoText: (data, _matches, output) => {
+        // No callout if missing numbers
+        const num = data.diveFromGraceNum[data.me];
+        if (!num) {
+          console.error(`DFG Tower 1 Reminder: missing number: ${JSON.stringify(data.diveFromGraceNum)}`);
+          return;
+        }
+        // Call 1st Tower Soak (Must be based on debuffs?)
+        if (num === 3) {
+          // Num3 High Jump Tower 1
+          if (data.diveFromGraceDir[data.me] === 'AC3') {
+            // Solo High Jump Tower 1
+            if (!data.diveFromGraceHasArrow[3])
+              return output.circleTower!();
+            // All High Jumps, unknown exact position
+            return output.circleTowers!({ num: output.num3!() });
+          }
+          // Num3 Spineshatter Tower 1
+          if (data.diveFromGraceDir[data.me] === 'AC4')
+            return output.spineshatterTower!();
+          // Num3 Elusive Tower 1
+          if (data.diveFromGraceDir[data.me] === 'AC5')
+            return output.elusiveTower!();
+        }
+      },
+      run: (data) => data.eyeOfTheTyrantCounter = 1,
+      outputStrings: {
+        num3: Outputs.num3,
+        circleTower: {
+          en: 'South Tower',
+        },
+        circleTowers: {
+          en: '${num} Cardinal Towers',
```

Comment:
Maybe just `Towers` or `Towers (all circles)` or `Get Tower`? I'm not sure the number part here adds a lot.

Reply:
I have changed it to Towers (all circles).